uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/document-list.controller.js

```diff
@@ -305,10 +290,10 @@ define([
         Array.prototype.push.apply(vm.list, output.documentList);
       });
 
-      // Calls addRemoveDocument function to add or remove the document in or from the list
+      // Adds new document in list or updates the existing document
       $scope.$on('documentFormSuccess', function (e, output, input) {
         if (angular.equals({}, input)) {
-          addRemoveDocument(vm.list, output, input);
+          vm.list.push(output);
```

Comment:
Looks like you are always adding the record, what will happen when a request is edited? will it add the same thing again and result in duplication?

Reply:
No it won't result in duplication as you can see here:
```js
// Adds new document in list or updates the existing document
$scope.$on('documentFormSuccess', function (e, output, input) {
    if (angular.equals({}, input)) {
       vm.list.push(output);
    } else {
          angular.extend(input, output);
    }
});
```
if `input` is empty the `output` will be added in the list. But in case the `input` is not empty, `output` object will be merges into `input`. SO it will not result in duplication.

Note: `input` => existing data and `output` => new data after its saved

Reply:
Could you please rename input with `existingRecord`/`existingData` and similarly for output, currently its confusing.